data_model/src/transaction.rs

```diff
@@ -337,10 +371,28 @@ impl TransactionValue {
         self.value.hash()
     }
 
-    /// [`Transaction`] payload.
+    /// Calculate transaction payload [`Hash`](`iroha_crypto::HashOf`).
+    #[cfg(feature = "std")]
+    pub fn hash_of_payload(&self) -> iroha_crypto::HashOf<TransactionPayload> {
```

Comment:
```suggestion
    #[cfg(feature = "std")]
    #[cfg(feature = "transparent_api")]
    pub fn hash_of_payload(&self) -> iroha_crypto::HashOf<TransactionPayload> {
```

Reply:
I suspect this method can even be removed. It's probably unused

Reply:
No, it's used

Reply:
It's used in the Client crate 
![image](https://github.com/hyperledger/iroha/assets/68690788/11138939-69c6-4da4-9484-6b3f2598a1aa)

Reply:
that's not the one defined on `TransactionValue`

Reply:
you can remove this one

Reply:
Ok. This one is defined in the SignedTransaction and is used:
![image](https://github.com/hyperledger/iroha/assets/68690788/db4a9169-913a-4b76-a826-a69dec732a1f)

And this one is defined in the TransactionValue and is used as well:
![image](https://github.com/hyperledger/iroha/assets/68690788/4ca7efbb-1ff2-48b9-a6cc-79398a6f0745)

All of these method outputs were tightly connected to a payload() method that was removed. So i can not remove any of newly created method without a deep refactoring all stable logic.

Reply:
Added a transparent API macro